spock-specs/src/test/groovy/org/spockframework/smoke/parameterization/InvalidWhereBlocks.groovy

```diff
@@ -268,4 +270,629 @@ c << $spock_p_a
+}
+"""
+
+    then:
+    InvalidSpecCompileException isce = thrown()
+    isce.message.startsWith('where-blocks may only contain parameterizations')
+
+    where:
+    dataTableSeparator << (2..20).collect { '_' * it }
+  }
+
+  def "different tables have to have same row count"() {
+    when:
+    runner.runFeatureBody """
+expect:
+true
+
+where:
+a | _
+1 | _
+4 | _
+3 | _
+__
+
+b | _
+2 | _
+5 | _
+"""
+
+    then:
+    SpockExecutionException see = thrown()
+    see.message =~ 'fewer values than previous'
+  }
+
+  def 'using combined label outside where block is not allowed'() {
+    when:
+    def result = compiler.transpileFeatureBody '''
+      combined:
+      true
+
+      expect:
+      true
+
+      where:
+      a | _
+      1 | _
+    '''
+
+    then:
+    result.source.normalize() == '''
+      |Unable to produce AST for this phase due to earlier compilation error:
+      |startup failed:
+      |script.groovy: 2: 'combined' is not allowed here; instead, use one of: [setup, given, expect, when, cleanup, where, end-of-method] @ line 2, column 7.
+      |         true
+      |         ^
+      |
+      |1 error
+    '''.stripMargin().trim()
+  }
+
+  def 'using combined label between data table and derived data variable is not allowed'() {
+    when:
+    def result = compiler.transpileFeatureBody '''
+      expect:
+      true
+
+      where:
+      a | _
+      1 | _
+      combined:
+      b = 1
+    '''
+
+    then:
+    result.source.normalize() == '''
+      |Unable to produce AST for this phase due to earlier compilation error:
+      |startup failed:
+      |script.groovy: 8: Derived data variables cannot be combined @ line 8, column 7.
+      |         b = 1
+      |         ^
+      |
+      |1 error
+    '''.stripMargin().trim()
+  }
+
+  def 'using combined label between derived data variable and data table is not allowed'() {
+    when:
+    def result = compiler.transpileFeatureBody '''
+      expect:
+      true
+
+      where:
+      a = 1
+      combined:
+      b | _
+      1 | _
+    '''
+
+    then:
+    result.source.normalize() == '''
+      |Unable to produce AST for this phase due to earlier compilation error:
+      |startup failed:
+      |script.groovy: 5: Derived data variables cannot be combined @ line 5, column 7.
+      |         a = 1
+      |         ^
+      |
+      |1 error
+    '''.stripMargin().trim()
+  }
+
+  def 'using combined label between data pipe and derived data variable is not allowed'() {
+    when:
+    def result = compiler.transpileFeatureBody '''
+      expect:
+      true
+
+      where:
+      a << [1]
+      combined:
+      b = 1
+    '''
+
+    then:
+    result.source.normalize() == '''
+      |Unable to produce AST for this phase due to earlier compilation error:
+      |startup failed:
+      |script.groovy: 7: Derived data variables cannot be combined @ line 7, column 7.
+      |         b = 1
+      |         ^
+      |
+      |1 error
+    '''.stripMargin().trim()
```

Comment:
How about an `assertErrorMessage(result, expected)` method,
wich does this:

```groovy
void assertErrorMessage(result, String expected){
   def actual = result.source.normalize()
   expected = expected.stripMargin().trim()
   assert actual != expected
}
```

This would reduce duplicated code and declutter the assert ouput, if the test is failing due to the  `.stripMargin().trim()` and other calls.